README.md

```diff
@@ -1,58 +1,110 @@
 
+**Stay Updated:**
+For upcoming features and community updates, please follow our project repository and join our discussions.
 
-## Status
+![Subnet Architecture](static/Subnet-Arch.png)
 
-We are on testnet, uid 168!
+We are currently in testnet phase, uid 168.
 
-[Join our Discord!](https://discord.gg/SaFbkGkU)
+## Join Our Discord Community
+
+<p align="left">
+  <a href="https://discord.gg/bitmind">
+    <img src="static/Join-BitMind-Discord.png" alt="Join us on Discord" width="60%">
+  </a>
+</p>
+
+For real-time discussions, community support, and regular updates, join our Discord server. Connect with developers, researchers, and users to get the most out of BitMind Subnet.
+
+## Project Structure and Terminology
+
+### Overview and Terminology
+
+Before diving into the specifics of the directory structure and key components, let's familiarize ourselves with the essential terms used throughout this project. Understanding these terms is crucial for navigating and contributing to the BitMind Subnet effectively. For a more detailed explanation of the terminology, please refer to [Bittensor Building Blocks](https://docs.bittensor.com/learn/bittensor-building-blocks).
+
+- **Synapse**: Acts as a communication bridge between axons (servers) and dendrites (clients), facilitating data flow and processing.
+- **Neuron**: A fundamental unit that includes both an axon and a dendrite, enabling full participation in the network operations.
+
+### Notable Directories
+
+- **bitmind/**: This directory contains the specific implementations of Bittensor operations, which include the key components such as miners, validators, and neurons.
+  - **base/**: Houses base classes for miner, validator, and neuron functionalities, each inheriting from the broader Bittensor framework.
+
+### Key Files and Descriptions
+
+#### bitmind/base/
+- **miner.py**: Responsible for loading models and weights, and handling predictions on images.
+- **validator.py**: Implements core functionality including reward calculation and forwarding operations.
+- **neuron.py**: A class that inherits from the base miner class provided by Bittensor, incorporating both axon and dendrite functionalities.
+
+#### bitmind/validator/
+- **forward.py**: Manages image processing and synapse operations using `ImageSynapse` for 256x256 images. Includes logic for challenge issuance and reward updates based on performance.
+- **proxy.py**: Temporarily unused; intended for handling frontend requests.
+
+#### bitmind/miner/
+- **predict.py**: Handles image transformation and the execution of model processing.
+
+### Utility and API Modules
+
+- **utils/**: Contains mostly boilerplate code necessary for general operations.
+- **api/**: Purely boilerplate, facilitating API interactions.
```

Comment:
lets actually remove this line along with this directory in this PR if you don't mind. This code is available in the bittensor-subnet-template repo if we ever need it in the future